tests/python/test_slowmo_fsdp.py

```diff
@@ -0,0 +1,378 @@
+        )
+
+        # create a subgroup equal to the whole WORLD
+        cur_subgroup = dist.distributed_c10d._get_default_group()
+        self.assertEqual(cur_subgroup.size(), dist.get_world_size())
+        slowmo_state = slow_momentum_comm.SlowMoState(cur_subgroup, sync_grads=False)
+        # check that state has subgroup registered
+        self.assertEqual(slowmo_state.subgroup.size(), cur_subgroup.size())
+
+        fsdp_net.register_comm_hook(slowmo_state, slow_momentum_comm.slowmo_hook)
+
+        # Make sure grads were not reduced, since `sync_grads` is set to False
+        # Gradient in this case is equal to rank
+        self._check_grads_eq_rank(fsdp_net, inpt)
+
+    @skip_if_lt_x_gpu(2)
+    @parametrize("sharding_strategy", [ShardingStrategy.NO_SHARD])
+    def test_slowmo_optimizer_averager(
+        self, sharding_strategy: Optional[ShardingStrategy]
+    ):
+        fsdp_net = self._init_fsdp(
+            sharding_strategy,
+            net=Net(has_wrapping=True, sharding_strategy=sharding_strategy),
+        )
+        fsdp_net_slowmo = self._init_fsdp(
+            sharding_strategy,
+            net=Net(has_wrapping=True, sharding_strategy=sharding_strategy),
+        )
+
+        cur_subgroup = dist.new_group(ranks=[self.rank])
+        slowmo_state = slow_momentum_comm.SlowMoState(cur_subgroup)
+        fsdp_net.register_comm_hook(slowmo_state, slow_momentum_comm.slowmo_hook)
+        fsdp_net_slowmo.register_comm_hook(slowmo_state, slow_momentum_comm.slowmo_hook)
+        inpt = torch.randn(  # type: ignore[call-overload]
+            (7, 8), dtype=torch.float, device=self.rank
+        )
+
+        slowmo_optim = self._get_slomo_optimizer(
+            base_optim=torch.optim.SGD(fsdp_net_slowmo.parameters(), lr=1e-2),
+            slowmo_freq=6,
+        )
+
+        # Manually changing slow momentum optimizer's averager's period
+        # to differ from `slowmo_freq` to check it independently from
+        # the momentum's update. Basically, parameter averaging now will happen
+        # every 3rd step and momentum step every 6th.
+        slowmo_optim.averager.period = 3
+
+        averager2 = self._get_averager(period=3)
+        base_optimizer = torch.optim.SGD(fsdp_net.parameters(), lr=1e-2)
+
+        for _ in range(4):
+            self._train_step(inpt, fsdp_net, base_optimizer)
+            self._train_step(inpt, fsdp_net_slowmo, slowmo_optim)
+            averager2.average_parameters(fsdp_net.parameters())
+
+        for slowmo_params, net_params in zip(
+            fsdp_net_slowmo.parameters(), fsdp_net.parameters()
+        ):
+            self.assertEqual(slowmo_params, net_params)
+
+    @skip_if_lt_x_gpu(2)
+    @parametrize("sharding_strategy", [ShardingStrategy.NO_SHARD])
+    def test_slowmo_optimizer_momentum_step(
+        self, sharding_strategy: Optional[ShardingStrategy]
+    ):
+        fsdp_net = self._init_fsdp(sharding_strategy)
```

Comment:
Might be helpful to add a comment here that this test assumes `fsdp_net` has a single top-level FSDP wrap